Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -37,11 +39,470 @@ public IEnumerator TestAddInteractableAtRuntime()
+            Assert.NotNull(defaultInputSource, "At least one input source must be present for this test to work.");
+
+            // Raise a voice select input event, then wait for transition to take place
+            SpeechCommands commands = new SpeechCommands("Select", KeyCode.None, interactable.InputAction);
+            MixedRealityToolkit.InputSystem.RaiseSpeechCommandRecognized(defaultInputSource, RecognitionConfidenceLevel.High, new System.TimeSpan(1000), System.DateTime.Now, commands);
+            yield return new WaitForSeconds(1f);
+
+            Assert.AreNotEqual(targetStartPosition, translateTargetObject.localPosition, "Transform target object was not translated by action.");
+
+            Object.Destroy(interactableObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+        }
+
+        /// <summary>
+        /// Instantiates a push button prefab and uses simulated global input events to press it.
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestSimulatedGlobalSelectInputOnPrefab()
+        {
+            // instantiate scene
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+            TestUtilities.InitializePlayspace();
+
+            // Load interactable prefab
+            GameObject interactableObject;
+            Interactable interactable;
+            Transform translateTargetObject;
+
+            // Place out of the way of any pointers
+            InstantiateDefaultInteractablePrefab(
+                new Vector3(10f, 0.0f, 0.5f),
+                new Vector3(-90f, 0f, 0f),
+                out interactableObject,
+                out interactable,
+                out translateTargetObject);
+
+            // Set interactable to global
+            interactable.IsGlobal = true;
+
+            Vector3 targetStartPosition = translateTargetObject.localPosition;
+
+            yield return null;
+
+            // Find an input source to associate with the input event (doesn't matter which one)
+            IMixedRealityInputSource defaultInputSource = MixedRealityToolkit.InputSystem.DetectedInputSources.FirstOrDefault();
+            Assert.NotNull(defaultInputSource, "At least one input source must be present for this test to work.");
+
+            // Add interactable as a global listener
+            MixedRealityToolkit.InputSystem.PushModalInputHandler(interactableObject);
+
+            // Raise a select down input event, then wait for transition to take place
+            MixedRealityToolkit.InputSystem.RaiseOnInputDown(defaultInputSource, Handedness.None, interactable.InputAction);
+            yield return new WaitForSeconds(1f);
+
+            Assert.AreNotEqual(targetStartPosition, translateTargetObject.localPosition, "Transform target object was not translated by action.");
```

Comment:
Seems like instead of making sure that the button moves, it would be better to register a delegate for the OnClick handler, like we do in pressablebuttontests:

```
bool buttonPressed = false;
buttonComponent.ButtonPressed.AddListener(() =>
{
    buttonPressed = true;
});
// press the button...
// blah
// blah
Assert.IsTrue(buttonPressed, "Button did not get pressed when hand moved to press it.");

```

Reply:
Note that this may require modifying Interactable.cs so that OnClick can actually be specified from code.